test/extensions/clusters/aggregate/cluster_test.cc

```diff
@@ -0,0 +1,221 @@
+#include "common/singleton/manager_impl.h"
+
+#include "extensions/clusters/aggregate/cluster.h"
+
+#include "test/common/upstream/utility.h"
+#include "test/mocks/protobuf/mocks.h"
+#include "test/mocks/server/mocks.h"
+#include "test/mocks/ssl/mocks.h"
+#include "test/test_common/environment.h"
+
+using testing::AtLeast;
+using testing::DoAll;
+using testing::Eq;
+using testing::InSequence;
+using testing::Return;
+using testing::ReturnRef;
+using testing::SizeIs;
+
+namespace Envoy {
+namespace Extensions {
+namespace Clusters {
+namespace Aggregate {
+
+class AggregateClusterTest : public testing::Test {
+public:
+  AggregateClusterTest() : stats_(Upstream::ClusterInfoImpl::generateStats(stats_store_)) {}
+
+  void setupPrioritySet() {
+    // Set up the HostSet with 1 healthy, 1 degraded and 1 unhealthy.
+    Upstream::HostSharedPtr host1 = Upstream::makeTestHost(info_, "tcp://127.0.0.1:80");
+    host1->healthFlagSet(Upstream::HostImpl::HealthFlag::DEGRADED_ACTIVE_HC);
+    Upstream::HostSharedPtr host2 = Upstream::makeTestHost(info_, "tcp://127.0.0.2:80");
+    host2->healthFlagSet(Upstream::HostImpl::HealthFlag::FAILED_ACTIVE_HC);
+    Upstream::HostSharedPtr host3 = Upstream::makeTestHost(info_, "tcp://127.0.0.3:80");
+
+    // Set up the HostSet with 2 healthy, 2 degraded and 2 unhealthy.
+    Upstream::HostSharedPtr host4 = Upstream::makeTestHost(info_, "tcp://127.0.0.4:80");
+    Upstream::HostSharedPtr host5 = Upstream::makeTestHost(info_, "tcp://127.0.0.5:80");
+    host4->healthFlagSet(Upstream::HostImpl::HealthFlag::DEGRADED_ACTIVE_HC);
+    host5->healthFlagSet(Upstream::HostImpl::HealthFlag::DEGRADED_ACTIVE_HC);
+    Upstream::HostSharedPtr host6 = Upstream::makeTestHost(info_, "tcp://127.0.0.6:80");
+    Upstream::HostSharedPtr host7 = Upstream::makeTestHost(info_, "tcp://127.0.0.7:80");
+    host6->healthFlagSet(Upstream::HostImpl::HealthFlag::FAILED_ACTIVE_HC);
+    host7->healthFlagSet(Upstream::HostImpl::HealthFlag::FAILED_ACTIVE_HC);
+    Upstream::HostSharedPtr host8 = Upstream::makeTestHost(info_, "tcp://127.0.0.8:80");
+    Upstream::HostSharedPtr host9 = Upstream::makeTestHost(info_, "tcp://127.0.0.9:80");
+
+    Upstream::HostVector hosts1{host1, host2, host3},
+        hosts2{host4, host5, host6, host7, host8, host9};
+    auto hosts1_ptr = std::make_shared<Upstream::HostVector>(hosts1);
+    auto hosts2_ptr = std::make_shared<Upstream::HostVector>(hosts2);
+
+    primary_ps_.updateHosts(
+        0,
+        Upstream::HostSetImpl::partitionHosts(hosts1_ptr, Upstream::HostsPerLocalityImpl::empty()),
+        nullptr, hosts1, {}, 100);
+
+    primary_ps_.updateHosts(
+        1,
+        Upstream::HostSetImpl::partitionHosts(hosts2_ptr, Upstream::HostsPerLocalityImpl::empty()),
+        nullptr, hosts2, {}, 100);
+
+    secondary_ps_.updateHosts(
+        0,
+        Upstream::HostSetImpl::partitionHosts(hosts2_ptr, Upstream::HostsPerLocalityImpl::empty()),
+        nullptr, hosts2, {}, 100);
```

Comment:
are you intentionally reusing the hosts lists between clusters? would making them unique make the tests more robust?